packages/entity-server/src/app/createApp.ts

```diff
@@ -36,9 +37,11 @@ export function createApp() {
   return new MicroServiceApiBuilder(new TupaiaDatabase())
     .useBasicBearerAuth('entity-server')
     .use<SingleEntityRequest | MultiEntityRequest>('hierarchy/:hierarchyName', attachCommonContext)
+    .use<MultiEntityRequest>('hierarchy/:hierarchyName$', attachMultiEntityContext)
     .use<MultiEntityRequest>('hierarchy/:hierarchyName/descendants', attachMultiEntityContext)
     .use<MultiEntityRequest>('hierarchy/:hierarchyName/relatives', attachMultiEntityContext)
     .use<MultiEntityRequest>('hierarchy/:hierarchyName/relationships', attachMultiEntityContext)
+    .post<MultiEntityRequest>('hierarchy/:hierarchyName$', handleWith(MultiEntityRoute))
```

Comment:
There are enough of these building up that it may get a little confusing which middlewares are attached, and someone adding a new one might miss the fact they should attach the middleware. There's a different pattern that I think could make everything more obvious, which is that in express you can generally call `.post` with middlewares inserted as params between the endpoint and the handler. It wouldn't be too hard to update the `ApiBuider` classes to allow that pattern too, and might make things like this a little more straightforward.

Out of official scope for this PR, so feel free to ignore, but let's both keep in mind for next time we do this sort of thing

Reply:
Love this suggestion! Will do 👍